src/type_traits/template_types.hpp

```diff
@@ -71,24 +71,27 @@ static RefVector<T> convertUPtrToRefVector(const UPtrVector<T>& ptr_list)
   return ref_list;
 }
 
-// temporary helper function
-template<class T>
-static RefVector<T> convertPtrToRefVector(const std::vector<T*>& ptr_list)
+/** convert a vector of std::unique_ptrs<T> to a refvector<T2>
+ *  the check for same is to prevent ambiguity between this function and the above
+ *  when T is a derived type of T2.
+ */
+template<class T2, class T, typename = std::enable_if_t<!std::is_same<T2, T>::value>>
```

Comment:
Can we have one function `template<class T, class T2 = T>`?

Reply:
I couldn't get that construction to work.  I have changed the selection to work via static assert which is more modern and cleaner.

Reply:
I'm confused. Before the change of this PR, did any compiler get confused by the two function templates?

Reply:
No we did not have code that hit the ambiguity.  But in expanding test coverage in my WIP PR I discovered a case where it is ambiguous.  If you try to run the new test code with the old `convertPtrToRefVector` definitions the compiler will have an error.

Reply:
test this please